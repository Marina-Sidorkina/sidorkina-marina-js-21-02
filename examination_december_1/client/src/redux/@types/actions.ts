import {
  IDummyApiResponse, IDummyUserFull, IDummyPostFull, IDummyComment
} from '../../api/proxy/@types/proxy';

export interface IAction {
  type: string
}

export interface ILoaderActionType extends IAction {
  payload: number;
}

export interface IRegistrationActionType extends IAction {
  payload: string;
}

export interface IUserInfoActionType extends IAction {
  payload?: IDummyUserFull;
  error?: any
}

export interface IUsersListActionType extends IAction {
  payload?: IDummyApiResponse;
  error?: any
}

export interface IUserPostsActionType extends IAction {
  payload?: IDummyApiResponse;
  error?: any
}

export interface IPostsActionType extends IAction {
  payload?: IDummyApiResponse;
  error?: any
}

export interface IPostModalPostActionType extends IAction {
  payload: IDummyPostFull;
  error?: any;
}

export interface IPostModalCommentsActionType extends IAction {
  payload: IDummyComment[],
  page: number,
  total: number
}

export interface IChangeLanguageActionType extends IAction {
  payload: string
}
