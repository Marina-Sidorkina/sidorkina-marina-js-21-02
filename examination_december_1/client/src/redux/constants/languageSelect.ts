export const CHANGE_LANGUAGE = 'LANGUAGE_SELECT/CHANGE_LANGUAGE';
