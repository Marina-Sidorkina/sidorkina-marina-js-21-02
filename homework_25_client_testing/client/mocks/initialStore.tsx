const initialStore = {
  userModal: {
    isOpened: false,
    values: {
      picture: '',
      name: 'John Doe',
      gender: 'male',
      dateOfBirth: {
        ruDateAndTime: '05 October 2011 14:48 UTC',
      },
      registerDate: {
        ruDateAndTime: '05 October 2011 14:48 UTC',
      },
      id: '12345',
      email: 'test',
      phone: '55555'
    }
  },
  login: {
    data: {
      error: false,
      isLoading: false,
      inputValue: '',
      authorizedUserId: '123345',
      authorizedUserName: 'Name',
      authorizedUserPicture: ''
    }
  },
  registration: {
    values: {
      name: '',
      gender: '',
      dateOfBirth: '',
      email: '',
      phone: '',
    },
    isLoading: false,
    error: false
  },
  languageSelector: {
    value: 'ru'
  },
  postModal: {
    isOpened: false,
    currentPostId: '12345'
  },
  userPosts: {
    data: {
      page: 0,
      total: 100,
      perPage: 6,
      posts: [{
        id: '12345',
        image: '',
        text: 'text',
        publishDate: {
          ruDateAndTime: '05 October 2011 14:48 UTC'
        },
        owner: {
          firstName: 'John',
          lastName: 'Doe',
          picture: '',
          title: 'mr',
          id: '12345'
        }
      }]
    }
  },
  postsList: {
    data: {
      page: 0,
      total: 100,
      perPage: 6,
      posts: [{
        id: '12345',
        image: '',
        text: 'text',
        publishDate: {
          ruDateAndTime: '05 October 2011 14:48 UTC'
        },
        owner: {
          firstName: 'John',
          lastName: 'Doe',
          picture: '',
          title: 'mr',
          id: '12345'
        }
      }]
    }
  },
  usersList: {
    data: {
      page: 0,
      total: 100,
      perPage: 6,
      users: [{
        firstName: 'John',
        lastName: 'Doe',
        title: 'mr',
        id: '12345'
      }]
    }
  },
  userInfo: {
    data: {
      isLoading: false,
      user: {
        firstName: 'John',
        lastName: 'Doe',
        picture: '',
        title: 'mr',
        gender: 'male',
        dateOfBirth: {
          ruDateAndTime: '05 October 2011 14:48 UTC',
        },
        registerDate: {
          ruDateAndTime: '05 October 2011 14:48 UTC',
        },
        id: '12345',
        email: '',
        phone: '55555'
      }
    }
  },
  postModalPost: {
    isLoading: false,
    error: false,
    owner: {
      firstName: 'John',
      lastName: 'Doe',
      picture: '',
      title: 'mr',
      id: '12345'
    },
    post: {
      id: '12345',
      image: '',
      text: 'text',
      publishDate: {
        ruDateAndTime: '05 October 2011 14:48 UTC'
      }
    }
  },
  postModalComments: {
    page: 1,
    error: false,
    isLoading: false,
    comments: [{
      id: '1234',
      message: 'message',
      publishDate: {
        ruDateAndTime: '05 October 2011 14:48 UTC'
      }
    }],
    owners: [{
      firstName: 'John',
      lastName: 'Doe',
      picture: '',
      title: 'mr',
      id: '12345'
    }],
    limit: 6,
    totalComments: 0
  }
};

export default initialStore;
