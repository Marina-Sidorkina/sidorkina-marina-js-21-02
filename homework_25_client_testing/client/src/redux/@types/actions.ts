import {
  IProxyApiResponse, IProxyUserFull, IProxyPostFull, IProxyComment
} from '../../api/proxy/@types/proxy';

export interface IAction {
  type: string
}

export interface ILoaderActionType extends IAction {
  payload: number;
}

export interface IRegistrationActionType extends IAction {
  payload: string;
}

export interface IUserInfoActionType extends IAction {
  payload?: IProxyUserFull;
  error?: any
}

export interface IUsersListActionType extends IAction {
  payload?: IProxyApiResponse;
  error?: any
}

export interface IUserPostsActionType extends IAction {
  payload?: IProxyApiResponse;
  error?: any
}

export interface IPostsActionType extends IAction {
  payload?: IProxyApiResponse;
  error?: any
}

export interface IPostModalPostActionType extends IAction {
  payload: IProxyPostFull;
  error?: any;
}

export interface IPostModalCommentsActionType extends IAction {
  payload: IProxyComment[],
  page: number,
  total: number
}

export interface IChangeLanguageActionType extends IAction {
  payload: string
}
