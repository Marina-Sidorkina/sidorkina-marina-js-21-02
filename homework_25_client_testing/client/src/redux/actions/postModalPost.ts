import { Dispatch } from 'redux';
import { AxiosResponse } from 'axios';
import {
  HIDE_POST_MODAL_POST_LOADING, SHOW_POST_MODAL_POST_LOADING, HIDE_POST_MODAL_POST_ERROR,
  SHOW_POST_MODAL_POST_ERROR, UPDATE_POST_MODAL_POST
} from '../constants/postModalPost';
import { getPostInfo } from '../../api/proxy';

export const showPostModalPostLoading = () => ({
  type: SHOW_POST_MODAL_POST_LOADING
});

export const hidePostModalPostLoading = () => ({
  type: HIDE_POST_MODAL_POST_LOADING
});

export const showPostModalPostError = () => ({
  type: SHOW_POST_MODAL_POST_ERROR
});

export const hidePostModalPostError = () => ({
  type: HIDE_POST_MODAL_POST_ERROR
});

export const updatePostModalPost = (post: AxiosResponse<any, any>) => ({
  type: UPDATE_POST_MODAL_POST,
  payload: post
});

export const getNewPostModalPost = (id: string) => (dispatch: Dispatch) => {
  dispatch(hidePostModalPostError());
  dispatch(showPostModalPostLoading());
  getPostInfo(id)
    .then((response) => {
      dispatch(hidePostModalPostLoading());
      dispatch(updatePostModalPost(response.data.data));
      dispatch(hidePostModalPostError());
    })
    .catch(() => {
      dispatch(hidePostModalPostLoading());
      dispatch(showPostModalPostError());
    });
};
