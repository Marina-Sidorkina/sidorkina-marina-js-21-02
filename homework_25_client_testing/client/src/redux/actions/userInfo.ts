import { AxiosResponse } from 'axios';
import { Dispatch } from 'redux';
import { getUserInfo } from '../../api/proxy';
import {
  LOAD_USER_INFO, SHOW_USER_INFO_LOADING, HIDE_USER_INFO_LOADING,
  HIDE_USER_INFO_ERROR, SHOW_USER_INFO_ERROR
} from '../constants/userInfo';

export const updateUserCardAction = (response: AxiosResponse<any, any>) => ({
  type: LOAD_USER_INFO,
  payload: response,
});

const showLoadingAction = () => ({
  type: SHOW_USER_INFO_LOADING,
});

const hideLoadingAction = () => ({
  type: HIDE_USER_INFO_LOADING,
});

export const showUserInfoErrorAction = () => ({
  type: SHOW_USER_INFO_ERROR
});

export const hideUserInfoErrorAction = () => ({
  type: HIDE_USER_INFO_ERROR
});

export const loadUserInfo = (paramsId: string) => (dispatch: Dispatch) => {
  dispatch(hideUserInfoErrorAction());
  dispatch(showLoadingAction());
  getUserInfo(paramsId)
    .then((response) => {
      dispatch(updateUserCardAction(response.data.data));
      dispatch(hideLoadingAction());
      dispatch(hideUserInfoErrorAction());
    })
    .catch(() => {
      dispatch(showUserInfoErrorAction());
      dispatch(hideLoadingAction());
    });
};
