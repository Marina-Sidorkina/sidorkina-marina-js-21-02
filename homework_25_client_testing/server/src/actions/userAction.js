const { updateUserById, createUser } = require("../api/dummyApi/index");
const logger = require('../logger');
const format = require('string-format');
const { userAction: messages } = require('../constants/loggerMessages');

class UserAction {
  updateUserById(id, data) {
    logger.info(format(messages.UPDATE_USER_BY_ID_INVOKE, id, JSON.stringify(data)));
    return updateUserById(id, data)
      .then((response) => {
        logger.info(format(messages.UPDATE_USER_BY_ID_SUCCESS, JSON.stringify(response.data)));
        return response.data;
      })
      .catch((error) => {
        logger.error(format(messages.UPDATE_USER_BY_ID_ERROR, JSON.stringify(error)));
        return Promise.reject(error);
      });
  }

  createUser(data) {
    logger.info(format(messages.CREATE_USER_INVOKE, JSON.stringify(data)));
    return createUser(data)
      .then((response) => {
        logger.info(format(messages.CREATE_USER_SUCCESS, JSON.stringify(response.data)));
        return response.data;
      })
      .catch((error) => {
        logger.error(format(messages.CREATE_USER_ERROR, JSON.stringify(error)));
        return Promise.reject(error);
      });
  }
}

module.exports = new UserAction();
