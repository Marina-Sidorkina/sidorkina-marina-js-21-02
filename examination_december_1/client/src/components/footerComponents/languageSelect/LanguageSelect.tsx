import React, { ChangeEvent } from 'react';
import { useDispatch } from 'react-redux';
import { useTypedSelector } from '../../../redux/hooks/useTypedSelector';
import { changeLanguageAction } from '../../../redux/actions/languageSelect';

const LanguageSelect = () => {
  const value = useTypedSelector((state) => state.languageSelector.value);
  const dispatch = useDispatch();

  const onLanguageChange = (evt: ChangeEvent<HTMLSelectElement>) => {
    dispatch(changeLanguageAction(evt.target.value));
  };

  return (
    <select
      className="language"
      name="language"
      id="language"
      value={value}
      onChange={(evt) => onLanguageChange(evt)}
    >
      <option value="ru">RU</option>
      <option value="en">EN</option>
    </select>
  );
};

export default LanguageSelect;
