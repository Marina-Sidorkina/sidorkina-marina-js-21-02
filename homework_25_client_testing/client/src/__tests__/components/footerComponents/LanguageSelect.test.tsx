import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { configure, mount } from 'enzyme';
import { Provider } from 'react-redux';
import mockStore from '../../../../mocks/mockStore';
import LanguageSelect from '../../../components/footerComponents/languageSelect/LanguageSelect';
import * as actions from '../../../redux/actions/languageSelect';
import initialStore from '../../../../mocks/initialStore';

jest.mock('../../../redux/actions/languageSelect');
configure({ adapter: new Adapter() });

describe('LanguageSelect component test', () => {
  let store: any;
  let wrapper: any;

  beforeEach(() => {
    store = mockStore(initialStore);
    store.dispatch = jest.fn();
    wrapper = mount(<Provider store={store}><LanguageSelect /></Provider>);
  });

  test('should render LanguageSelect', () => {
    expect(wrapper.length).toEqual(1);
  });

  test('prop should match with store', () => {
    expect(wrapper.prop('languageSelector')).toEqual(store.languageSelector);
  });

  test('should render LanguageSelect', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should change value', () => {
    wrapper.childAt(0).simulate('change', { target: { value: 'en' } });
    expect(actions.changeLanguageAction).toBeCalledWith('en');
  });
});
