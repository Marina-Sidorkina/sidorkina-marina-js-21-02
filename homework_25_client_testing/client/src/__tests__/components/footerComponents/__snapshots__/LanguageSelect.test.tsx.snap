// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelect component test should render LanguageSelect 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <LanguageSelect>
    <select
      className="language"
      id="language"
      name="language"
      onChange={[Function]}
      value="ru"
    >
      <option
        value="ru"
      >
        RU
      </option>
      <option
        value="en"
      >
        EN
      </option>
    </select>
  </LanguageSelect>
</Provider>
`;
